2019-02-07  Brian Patrick Tagle  <tagle@lnxsrv09.seas.ucla.edu>

        * timespec.h: replaced integer subtraction in the timespec_cmp
        function with direct comparison with > and <.  If the seconds are equal
        then the nanoseconds are compared (this comparison still has the
        possibility of overflow).

--- lib/timespec.h      2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h      2019-02-07 10:17:06.355996000 -0800
@@ -45,8 +45,8 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  return a.tv_sec > b.tv_sec ? 1 : a.tv_sec < b.tv_sec ? -1 :
+    a.tv_nsec -b.tv_nsec;
 }

 # if ! HAVE_DECL_NANOSLEEP
